src/fides/api/ops/schemas/messaging/messaging.py

```diff
@@ -84,6 +87,42 @@ class RequestReviewDenyBodyParams(BaseModel):
     rejection_reason: Optional[str]
 
 
+class ConsentPreferencesByUser(BaseModel):
+    """Used for capturing the preferences of a single user.
+
+    Used for ConsentEmailFulfillmentBodyParams where we potentially send a list
+    of batched user preferences to a third party vendor all at once.
+    """
+
+    identities: Dict[str, Any]
+    consent_preferences: List[Consent]  # Consent schema
+
+    @root_validator
+    def transform_data_use_format(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a data use fides_key to a corresponding name if possible"""
+        consent_preferences = values.get("consent_preferences") or []
+        for preference in consent_preferences:
+            preference.data_use = next(
+                (
+                    data_use.name
+                    for data_use in DEFAULT_TAXONOMY.data_use
+                    if data_use.fides_key == preference.data_use
+                ),
+                preference.data_use,
+            )
+        values["consent_preferences"] = consent_preferences
+        return values
```

Comment:
Consent preferences are saved under a fides key (e.g. advertising.first_party), so here we attempt to map to its `name` instead before sending in the email